pallets/pallet-governance/src/lib.rs

```diff
@@ -0,0 +1,446 @@
+							id: proposal,
+							call: call.clone(),
+							result: Self::dispatch_governance_call(call),
+						});
+					} else {
+						Self::deposit_event(Event::<T>::ProposalRejected {
+							id: proposal,
+							reason: RejectReason::ByVoting,
+						});
+					}
+				}
+				// Deleted the related data in Votes.
+				Votes::<T>::remove(proposal);
+			});
+
+			// Take the expiry and clean up the storage if any proposal expired and log an event.
+			Expiry::<T>::take(block_number).into_iter().for_each(|proposal_id| {
+				// The resolved proposals should be cleaned up already, in this case, only log an
+				// event with the expired proposal.
+				if Proposals::<T>::contains_key(proposal_id) {
+					Self::deposit_event(Event::<T>::ProposalRejected {
+						id: proposal_id,
+						reason: RejectReason::Expired,
+					});
+				}
+
+				Proposals::<T>::remove(proposal_id);
+				Votes::<T>::remove(proposal_id);
+			});
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Registered a proposal.
+		///
+		/// Require a member from council.
+		#[pallet::call_index(0)]
+		#[pallet::weight(T::WeightInfo::initiate_proposal())]
+		pub fn initiate_proposal(
+			origin: OriginFor<T>,
+			call: Box<<T as Config>::RuntimeCall>,
+		) -> DispatchResult {
+			let id = ensure_signed(origin)?;
+			ensure!(CurrentAuthorities::<T>::get().contains(&id), Error::<T>::Unauthorized);
+			let proposal_id = Self::next_proposal_id();
+			let expired_block =
+				T::BlockNumberProvider::current_block_number() + T::ExpiryPeriod::get();
+
+			// add call to storage
+			Proposals::<T>::insert(proposal_id, call.encode());
+
+			// add proposal id to expiry.
+			// Get the BTreeSet for the block number and add the proposal id into the set.
+			Expiry::<T>::mutate(expired_block, |expiry_set| {
+				// Insert the proposal id into the set
+				expiry_set.insert(proposal_id);
+			});
+
+			// add vote to Votes.
+			let _ = Self::do_vote(id.clone(), proposal_id, true);
+
+			// Emit Event: ProposalRegistered
+			Self::deposit_event(Event::<T>::ProposalRegistered { who: id, call: *call });
+
+			Ok(())
+		}
+
+		/// Allows council members to cast their votes (approval or rejection) on a given proposal.
+		/// It ensures that votes are cast by authorized members.
+		/// Checks the validity of the proposal, handles voting, determines if the proposal is ready
+		/// for resolution.
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::vote())]
+		pub fn vote(origin: OriginFor<T>, proposal: ProposalId, approve: bool) -> DispatchResult {
+			// check if the vote can be casted
+			// is member of the council
+			let id = ensure_signed(origin)?;
+			ensure!(CurrentAuthorities::<T>::get().contains(&id), Error::<T>::Unauthorized);
+
+			// Proposal is valid
+			ensure!(Proposals::<T>::contains_key(proposal), Error::<T>::InvalidProposalId);
+
+			// Vote
+			Self::do_vote(id.clone(), proposal, approve)
+		}
+
+		/// Requires Root origin - sets the current authorities
```

Comment:
We also extend voting period and clear old votes from current authority. Should add these to the documentation here and below